src/frontend/components/view--goal-search.tsx

```diff
@@ -12,19 +13,62 @@ interface ViewGoalSearch {
   total: number
 }
 
+async function getData(fulltext: string) {
+  const url = `/api/goal-search?fulltext=${fulltext}`;
+  try {
+    const response = await fetch(url);
+    if (!response.ok) {
+      throw new Error(`Response status: ${response.status}`);
+    }
+    
+    const { data } = await response.json();
+    console.log(data);
+    return {
+      goals: data?.goalsGraphql1?.results ?? [],
+      filters: data?.goalsGraphql1?.filters ?? [],
+      total: data?.goalsGraphql1?.pageInfo?.total ?? 0,
+      description: data?.goalsGraphql1?.description ?? "",
+    };
+  } catch (error) {
+    console.error(error.message);
+  }
+  
+}
+
 export default function GoalsSearchView({ filters, goals, total, description }: ViewGoalSearch) {
-  const [fulltext, setFulltext] = useState(filters[0].value ? filters[0].value : "")
-  const [facets] = useState([...Object.keys(filters[1].options)])
+  const [fulltext, setFulltext] = useState(filters[0]?.value ? filters[0].value : "")
+  const [displayGoals, setDisplayGoals] = useState(goals)
+  const [facets] = useState(filters[1] ? [...Object.keys(filters[1]?.options)] : [])
   const [search, setSearch] = useState(false);
   const [offset, setOffset] = useState(9);
   const [activeTopics, setActiveTopics] = useState([])
   const [notDisabledTopics, setNotDisabledTopics] = useState([])
   const [filteredGoals, setFilteredGoals] = useState([...goals])
   const [filteredGoalsCount, setFilteredGoalsCount] = useState(total)
 
-  function handleSearch(e: FormEvent) {
+  async function handleSearch(e: FormEvent) {
     e.preventDefault()
-    setSearch(true)
+    const url = `/api/goal-search?fulltext=${fulltext}`;
```

Comment:
do we need to sanitize this input at all? this is pulling from the text field users type into rather than the tag cloud right?

Reply:
I think we are fine without sanitizing it. Drupal is ultimately getting the response and will be running its own sanitization on the string. Usually I am more worried about sanitizing the output on the pages, which we might want to add. We don't sanitize anything going into DKAN from the React code, but let the processors in PHP handle it and log anything weird.